meos/src/general/temporal_analytics.c

```diff
@@ -273,7 +273,9 @@ tsequence_tprecision(const TSequence *seq, const Interval *duration,
       {
         start = end; end = NULL;
       }
-      if (interp != DISCRETE)
+      if (interp == DISCRETE)
+        k = 0;
+      else
```

Comment:
Does this not change the tests? If not, we should probably add a couple that test this condition.